relation.go

```diff
@@ -7,39 +7,62 @@ import (
 	"time"
 )
 
+// Comparator represents a comparison operand
+type Comparator int
+
 const (
-	equality = iota
-	in
-	greaterThan
-	greaterThanOrEquals
-	lesserThan
-	lesserThanOrEquals
+	// These comparison types represent the comparison types supported
+	// when generating a relation between a field and it's terms
+	CmpEquality            Comparator = iota // direct equality (foo = bar)
+	CmpIn                                    // membership (foo IN (bar, bing, baz))
+	CmpGreaterThan                           // larger than (foo > 1)
+	CmpGreaterThanOrEquals                   // larger than or equal (foo >= 1)
+	CmpLesserThan                            // less than (foo < 1)
+	CmpLesserThanOrEquals                    // less than or equal (foo <= 1)
 )
 
+// Relation describes the comparison of a field against a list of terms
+// that need to satisfy a comparator
 type Relation struct {
-	op    int
-	key   string
+	cmp   Comparator
+	field string
 	terms []interface{}
 }
 
+// Field provides the field name for this relation
+func (r Relation) Field() string {
```

Comment:
Similar thing here – I wonder if it would be simpler to just expose the fields rather than have accessors. Again though, I don't think it matters that much/I do not feel strongly.